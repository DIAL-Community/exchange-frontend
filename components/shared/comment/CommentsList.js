import { useCallback, useEffect } from 'react'
import { useIntl } from 'react-intl'
import { Loading } from '../FetchStatus'
import CommentCard from './CommentCard'

const CommentsList = ({ comments, refetch, loading, onClose }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id) => formatMessage({ id }), [formatMessage])

  useEffect(
    () => {
      refetch
    },
    [refetch]
  )

  return (
    <div className='mt-28'>
      <div className='card-title text-dial-gray-dark mb-3'>
        {format('app.comment')}
      </div>
      <div className='bg-edit p-6'>
        {loading ? <Loading /> : (
          comments?.length ? (
            <div className='flex flex-col gap-3'>
              {comments?.map(({ comId, fullName, avatarUrl, text, replies }, commentIdx) => (
                <CommentCard
                  key={commentIdx}
                  commentId={comId}
                  authorFullName={fullName}
                  authorAvatarUrl={avatarUrl}
                  text={text}
                  replies={replies}
                />
              ))}
            </div>
          ) : format('shared.comment.no-comments')
        )}
        <div className='flex justify-end mt-8 gap-3 text-xl'>
          <button
            type='button'
            onClick={onClose}
            className='cancel-button'
            data-testid='close-button'
          >
            {format('app.close')}
          </button>
        </div>
      </div>
    </div>
  )
}

export default CommentsList
