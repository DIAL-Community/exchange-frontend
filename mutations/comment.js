import { gql } from '@apollo/client'

export const CREATE_COMMENT = gql`
  mutation CreateComment(
    $commentId: String!
    $commentObjectType: String!
    $commentObjectId: Int!
    $text: String!
    $userId: Int!
    $parentCommentId: String
  ) {
    createComment(
      commentId: $commentId
      commentObjectType: $commentObjectType
      commentObjectId: $commentObjectId
      text: $text
      userId: $userId
      parentCommentId: $parentCommentId
    ) {
      errors
    }
  }
`
export const DELETE_COMMENT = gql`
  mutation DeleteComment($commentId: String!) {
    deleteComment(commentId: $commentId) {
      errors
    }
  }
`
