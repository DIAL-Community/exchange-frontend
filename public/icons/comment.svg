<svg xmlns="http://www.w3.org/2000/svg" width="27.222" height="20.94" viewBox="0 0 27.222 20.94">
    <defs>
        <style>
            .cls-1{fill:#3f9edd}
        </style>
    </defs>
    <g id="Group_1034" transform="translate(-9.375 -18.75)">
        <path id="Path_557" d="M29.188 45.842a2.094 2.094 0 0 1-2.094 2.094A2.094 2.094 0 0 1 25 45.842a2.094 2.094 0 1 1 4.188 0z" class="cls-1" transform="translate(-10.39 -16.622)"/>
        <path id="Path_558" d="M47.938 45.842a2.094 2.094 0 0 1-2.094 2.094 2.094 2.094 0 0 1-2.094-2.094 2.094 2.094 0 1 1 4.188 0z" class="cls-1" transform="translate(-22.858 -16.622)"/>
        <path id="Path_559" d="M66.688 45.842a2.094 2.094 0 0 1-2.094 2.094 2.094 2.094 0 0 1-2.094-2.094 2.094 2.094 0 0 1 4.188 0z" class="cls-1" transform="translate(-35.326 -16.622)"/>
        <path id="Path_560" d="M22.985 18.75a15.891 15.891 0 0 0-9.512 2.969 9.486 9.486 0 0 0-4.1 7.5 9.486 9.486 0 0 0 4.1 7.5 15.891 15.891 0 0 0 9.512 2.969h12.564a1.047 1.047 0 0 0 .74-1.787l-2.409-2.415a8.778 8.778 0 0 0 2.72-6.266 9.488 9.488 0 0 0-4.1-7.5 15.886 15.886 0 0 0-9.515-2.97zm8.255 4.644a7.349 7.349 0 0 1 3.26 5.826c0 2.147-.935 3.788-2.82 5.474a1.047 1.047 0 0 0-.041 1.521l1.38 1.38H22.985a13.877 13.877 0 0 1-8.257-2.552 7.343 7.343 0 0 1-3.26-5.824 7.346 7.346 0 0 1 3.26-5.826 13.877 13.877 0 0 1 8.257-2.55 13.871 13.871 0 0 1 8.255 2.55z" class="cls-1"/>
    </g>
</svg>
