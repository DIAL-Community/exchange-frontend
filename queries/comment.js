import { gql } from '@apollo/client'

export const COMMENTS_QUERY = gql`
  query Comments(
    $commentObjectType: String!
    $commentObjectId: Int!
  ) {
    comments(
      commentObjectType: $commentObjectType
      commentObjectId: $commentObjectId
    ) {
      userId
      comId
      fullName
      avatarUrl
      text
      replies {
        userId
        comId
        fullName
        avatarUrl
        text
      }
    }
  }
`
