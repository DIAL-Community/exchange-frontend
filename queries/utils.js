const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
