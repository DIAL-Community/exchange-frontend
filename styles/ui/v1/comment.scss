.szh-menu {
  @apply border;
  @apply bg-white;
  @apply shadow-md;
  @apply text-dial-stratos;
}

/* Delete confirmation */
.react-responsive-modal-modal {
  @apply p-8;
  @apply w-1/2;
  @apply rounded-2xl;
  @apply max-w-prose;

  h2 {
    @apply pb-3;
    @apply text-lg;
    @apply font-semibold;
    @apply text-dial-sapphire;
    text-align: left !important;
  }

  p {
    @apply pb-3;
    @apply text-sm;
    text-align: left !important;
  }

  .deleteBtns {
    justify-content: flex-end;

    button.delete {
      @apply px-5;
      @apply py-2;
      @apply font-normal;
    }

    button.cancel {
      @apply px-5;
      @apply py-2;
      @apply font-normal;
    }
  }
}

/* CommentsSection - wrapper */
#comments-section {
  @apply pt-20;

  .overlay {
    @apply p-0;

    .replySection {
      @apply border-l-2;
      @apply border-dial-slate-300;
    }

    .advanced-overlay {
      @apply gap-x-2;

      .userImg {
        @apply w-12;
        @apply h-12;
      }
    }

    .comment-title {
      @apply text-lg;
      @apply text-dial-sapphire;
      @apply py-2;
    }

    /* CommentsSection - no comments */
    .no-comDiv {
      @apply my-8;
      @apply text-sm;
      @apply font-normal;
    }

    .signBox {
      @apply mt-3;
      @apply p-3;
      @apply items-center;

      .signLine {
        @apply text-sm;
        @apply font-normal;
        @apply m-0;
      }

      button {
        @apply py-2.5;
        @apply px-5;
        @apply border-0;
        @apply rounded-md;
        @apply text-white;
        @apply hover:border-0;
        @apply text-sm;
        @apply font-normal;
      }
    }

    .replyBtn {
      @apply text-sm;
      @apply font-normal;
      @apply tracking-normal;

      .replyIcon {
        @apply mt-[2px];
      }
    }

    button.loginBtn,
    button.signBtn {
      @apply bg-dial-sapphire;
      @apply disabled:bg-dial-gray;
      @apply hover:bg-dial-sapphire;
      @apply hover:text-white;
    }
  }

  /* CommentsSection - text input */
  .form {
    .advanced-border {
      @apply border;
      @apply text-dial-gray-dark;
      @apply shadow-md;
      @apply rounded-md;
    }

    /* CommentsSection - Post and Cancel buttons */
    .advanced-btns {
      @apply mt-3;
      @apply flex-row-reverse;
      @apply gap-3;

      button {
        @apply flex;
        @apply items-center;
        @apply py-2;
        @apply px-5;
        @apply border-0;
        @apply rounded-md;
        @apply text-white;
        @apply text-sm;
        @apply disabled:cursor-not-allowed;
        @apply hover:border-0;
        @apply font-normal;

        &.advanced-post {
          @apply bg-dial-sapphire;
          @apply disabled:bg-dial-slate-300;
        }

        &.advanced-cancel {
          @apply bg-button-gray-light;
          @apply mr-0;
        }
      }
    }
  }

  /* CommentsSection - Edit and Delete buttons */
  .overlay .userActions .szh-menu__item {
    @apply capitalize;
  }
}
