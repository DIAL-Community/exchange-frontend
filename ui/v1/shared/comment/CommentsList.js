import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { Loading } from '../FetchStatus'
import CommentCard from './CommentCard'

const CommentsList = ({ comments, loading, onClose }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id) => formatMessage({ id }), [formatMessage])

  return (
    <div className='mt-28'>
      <div className='card-title text-dial-gray-dark mb-3'>
        {format('app.comment')}
      </div>
      <div className='bg-edit p-6'>
        {loading ? <Loading /> : (
          comments?.length ? (
            <div className='flex flex-col gap-3'>
              {comments?.map(({ comId, fullName, avatarUrl, text, replies }, commentIdx) => (
                <CommentCard
                  key={commentIdx}
                  commentId={comId}
                  authorFullName={fullName}
                  authorAvatarUrl={avatarUrl}
                  text={text}
                  replies={replies}
                />
              ))}
            </div>
          ) : format('shared.comment.no-comments')
        )}
        <div className='flex justify-end mt-8 gap-3 text-xl'>
          <button
            type='button'
            onClick={onClose}
            className='cancel-button text-sm'
          >
            {format('app.close')}
          </button>
        </div>
      </div>
    </div>
  )
}

export default CommentsList
