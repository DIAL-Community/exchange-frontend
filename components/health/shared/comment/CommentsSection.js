import { useCallback, useEffect, useRef, useState } from 'react'
import classNames from 'classnames'
import dynamic from 'next/dynamic'
import { useRouter } from 'next/router'
import { useMutation, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../../lib/apolloClient'
import { useUser } from '../../../../lib/hooks'
import CommentsList from '../../../shared/comment/CommentsList'
import { Loading } from '../../../shared/FetchStatus'
import EditButton from '../../../shared/form/EditButton'
import { CREATE_COMMENT, DELETE_COMMENT } from '../../../shared/mutation/comment'
import { COMMENTS_COUNT_QUERY, COMMENTS_QUERY } from '../../../shared/query/comment'

const CommentSection = dynamic(
  () => import('react-comments-section').then((module) => module.CommentSection),
  { ssr: false }
)

const INPUT_CLASSNAME = 'public-DraftStyleDefault-block'
const INPUT_BORDERED_WRAPPER_CLASSNAME = 'advanced-border'
const FOCUSED_CLASSNAME = 'focused'
const FIRST_ELEMENT_INDEX = 0

const UI_AVATAR_PARAMS = 'background=2e3192&color=fff&format=svg'

const CommentsSection = ({ objectId, objectType, commentsSectionRef, className }) => {
  const { locale } = useRouter()
  const innerRef = useRef(null)
  const { loadingUserSession, user, isAdminUser } = useUser()

  const { data, loading } = useQuery(COMMENTS_QUERY, {
    variables: {
      commentObjectId: parseInt(objectId),
      commentObjectType: objectType
    },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  const [createComment] = useMutation(CREATE_COMMENT, {
    refetchQueries: [{
      query: COMMENTS_COUNT_QUERY,
      variables: {
        commentObjectId: parseInt(objectId),
        commentObjectType: objectType
      },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }, {
      query: COMMENTS_QUERY,
      variables: {
        commentObjectId: parseInt(objectId),
        commentObjectType: objectType
      },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }]
  })
  const [deleteComment] = useMutation(DELETE_COMMENT, {
    refetchQueries: [{
      query: COMMENTS_COUNT_QUERY,
      variables: {
        commentObjectId: parseInt(objectId),
        commentObjectType: objectType
      },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }, {
      query: COMMENTS_QUERY,
      variables: {
        commentObjectId: parseInt(objectId),
        commentObjectType: objectType
      },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }]
  })

  const onCommentUpsertAction = (text, commentId, parentCommentId = null, parentOfRepliedCommentId = null) => {
    if (user) {
      createComment({
        variables: {
          commentId,
          commentObjectType: objectType,
          commentObjectId: parseInt(objectId),
          userId: user.id,
          parentCommentId: parentOfRepliedCommentId ?? parentCommentId,
          text
        },
        context: {
          headers: {
            'Accept-Language': locale
          }
        }
      })
    }
  }

  const onCommentDeleteAction = (commentId) => {
    if (user) {
      deleteComment({
        variables: { commentId },
        context: {
          headers: {
            'Accept-Language': locale
          }
        }
      })
    }
  }

  const getElementsByClassName = useCallback((className) => {
    return Array.from(innerRef?.current?.getElementsByClassName(className) ?? [])
  },[])

  const handleClickOutside = useCallback(({ target }) => {
    const inputs = getElementsByClassName(INPUT_BORDERED_WRAPPER_CLASSNAME)
    if (!inputs.some(element => element.contains(target))) {
      inputs.forEach(input => input.classList.remove(FOCUSED_CLASSNAME))
    }
  }, [getElementsByClassName])

  useEffect(() => {
    document.addEventListener('click', handleClickOutside)

    return () => {
      document.removeEventListener('click', handleClickOutside, false)
    }
  }, [handleClickOutside])

  const focusActiveElement = () => {
    const activeInput = document.activeElement.getElementsByClassName(INPUT_CLASSNAME)[FIRST_ELEMENT_INDEX]
    const focusedInputIndex = getElementsByClassName(INPUT_CLASSNAME).findIndex(input => input === activeInput)
    getElementsByClassName(INPUT_BORDERED_WRAPPER_CLASSNAME).forEach((input, inputIdx) => {
      if (inputIdx === focusedInputIndex) {
        input.classList.add(FOCUSED_CLASSNAME)
      } else {
        input.classList.remove(FOCUSED_CLASSNAME)
      }
    })
  }

  const [commentData, setCommentData] = useState([])

  useEffect(() => {
    if (data?.comments) {
      const commentData = []
      data.comments.forEach(({ replies, ...otherCommentProps }) => {
        commentData.push({
          replies: replies.map(({ ...otherReplyProps }) => otherReplyProps),
          ...otherCommentProps
        })
      })
      setCommentData(commentData)
    }
  }, [data?.comments])

  const [isInEditMode, setIsInEditMode] = useState(false)

  const toggleIsInEditMode = () => setIsInEditMode(!isInEditMode)

  return (
    <div ref={commentsSectionRef} className={classNames(className, 'text-dial-sapphire max-w-[50vw]')}>
      {loadingUserSession && <Loading />}
      {isInEditMode
        ? <CommentsList
          comments={data?.comments}
          loading={loading}
          onClose={toggleIsInEditMode}
          objectId={objectId}
          objectType={objectType}
        />
        : (
          <div id='comments-section' ref={innerRef} onClick={focusActiveElement}>
            {isAdminUser && (
              <div className='flex justify-end'>
                <EditButton onClick={toggleIsInEditMode} />
              </div>
            )}
            <CommentSection
              commentData={commentData}
              currentUser={user ? {
                currentUserId: String(user.id),
                currentUserImg: `https://ui-avatars.com/api/name=${user.userName}&${UI_AVATAR_PARAMS}`,
                currentUserFullName: user.userName
              } : null}
              logIn={{
                loginLink: '/auth/signin',
                signupLink: '/auth/signup'
              }}
              onSubmitAction={({ text, comId }) => onCommentUpsertAction(text, comId)}
              onReplyAction={
                ({ text, comId, repliedToCommentId, parentOfRepliedCommentId }) =>
                  onCommentUpsertAction(text, comId, repliedToCommentId, parentOfRepliedCommentId)
              }
              onEditAction={({ text, comId }) => onCommentUpsertAction(text, comId)}
              onDeleteAction={({ comIdToDelete }) => onCommentDeleteAction(comIdToDelete)}
              advancedInput
              removeEmoji
              hrStyle={{ borderColor: '#dfdfea' }}
              formStyle={{ backgroundColor: 'white', fontFamily: 'Poppins, sans-serif' }}
              imgStyle={{ margin: 'auto' }}
            />
          </div>
        )
      }
    </div>
  )
}

export default CommentsSection
