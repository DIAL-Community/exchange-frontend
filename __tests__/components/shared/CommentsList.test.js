import { fireEvent } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import CommentsList from '../../../components/shared/comment/CommentsList'
import { DELETE_COMMENT } from '../../../mutations/comment'
import { mockObserverImplementation, render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import { comments, deleteCommentFailure, deleteCommentSuccess } from './data/CommentsList'

mockNextUseRouter()
describe('Unit test for the CommentsList component.', () => {
  const DELETE_BUTTON_TEST_ID = 'delete-button'
  const CONFIRM_ACTION_DIALOG_TEST_ID = 'confirm-action-dialog'
  const CONFIRM_BUTTON_TEST_ID = 'confirm-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const mockOnClose = jest.fn()
  const mockRefetch = jest.fn()

  beforeAll(() => {
    window.ResizeObserver = mockObserverImplementation()
    window.IntersectionObserver = mockObserverImplementation()
  })

  describe('Should match snapshot', () => {
    test('loading comments.', () => {
      const { container } = render(
        <CommentsList
          comments={comments}
          refetch={mockRefetch}
          loading={true}
          onClose={mockOnClose}
        />
      )
      expect(container).toMatchSnapshot()
    })

    test('without comments.', () => {
      const { container } = render(
        <CommentsList
          comments={[]}
          refetch={mockRefetch}
          loading={false}
          onClose={mockOnClose}
        />
      )
      expect(container).toMatchSnapshot()
    })

    test('with comments.', () => {
      const { container } = render(
        <CustomMockedProvider addTypename={false}>
          <CommentsList
            comments={comments}
            refetch={mockRefetch}
            loading={false}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      expect(container).toMatchSnapshot()
    })
  })

  test('Should open ConfirmActionDialog after user clicks "Delete" button.', () => {
    const { getByTestId, getAllByTestId } = render(
      <CustomMockedProvider addTypename={false}>
        <CommentsList
          comments={comments}
          refetch={mockRefetch}
          loading={false}
          onClose={mockOnClose}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getAllByTestId(DELETE_BUTTON_TEST_ID)[0])
    expect(getByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toBeVisible()
    expect(getByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toMatchSnapshot()
  })

  test('Should close ConfirmActionDialog after user clicks "Cancel" button.', async () => {
    const { getByTestId, getAllByTestId, queryByTestId } = render(
      <CustomMockedProvider addTypename={false}>
        <CommentsList
          comments={comments}
          refetch={mockRefetch}
          loading={false}
          onClose={mockOnClose}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getAllByTestId(DELETE_BUTTON_TEST_ID)[0])
    expect(getByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toBeVisible()
    await act(async () => fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID)))
    expect(queryByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toBeNull()
  })

  describe('Should delete a comment after user clicks "Confirm" button', () => {
    test('successfully.', async () => {
      const mockDeleteComment = generateMockApolloData(
        DELETE_COMMENT,
        { commentId: '100' },
        null,
        deleteCommentSuccess
      )
      const { getByTestId, getAllByTestId, findByText } = render(
        <CustomMockedProvider mocks={[mockDeleteComment]} addTypename={false}>
          <CommentsList
            comments={comments}
            refetch={mockRefetch}
            loading={false}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      fireEvent.click(getAllByTestId(DELETE_BUTTON_TEST_ID)[0])
      await act(async () => fireEvent.click(getByTestId(CONFIRM_BUTTON_TEST_ID)))
      expect(await findByText('Comment deleted successfully')).toBeInTheDocument()
    })

    test('with failure', async () => {
      const mockDeleteComment = generateMockApolloData(
        DELETE_COMMENT,
        { commentId: '100' },
        new Error('An error occurred'),
        deleteCommentFailure
      )
      const { getByTestId, getAllByTestId, findByText } = render(
        <CustomMockedProvider mocks={[mockDeleteComment]} addTypename={false}>
          <CommentsList
            comments={comments}
            refetch={mockRefetch}
            loading={false}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      fireEvent.click(getAllByTestId(DELETE_BUTTON_TEST_ID)[0])
      await act(async () => fireEvent.click(getByTestId(CONFIRM_BUTTON_TEST_ID)))
      expect(await findByText('Comment deletion failed')).toBeInTheDocument()
    })
  })
})
