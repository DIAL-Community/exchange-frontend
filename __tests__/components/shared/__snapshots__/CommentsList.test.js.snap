// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the CommentsList component. Should match snapshot loading comments. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-28"
  >
    <div
      class="card-title text-dial-gray-dark mb-3"
    >
      Comments
    </div>
    <div
      class="bg-edit p-6"
    >
      <div
        class="h-[60vh]"
      >
        <div
          class="flex bg-dial-alice-blue h-full"
        >
          <div
            class="text-button-gray text-lg w-full my-auto flex flex-col gap-4"
          >
            <svg
              class="spinner mx-auto"
              fill="currentColor"
              height="3em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="3em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
              />
            </svg>
            <div
              class="text-center mt-5"
            >
              Fetching data ...
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="cancel-button"
          data-testid="close-button"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the CommentsList component. Should match snapshot with comments. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-28"
  >
    <div
      class="card-title text-dial-gray-dark mb-3"
    >
      Comments
    </div>
    <div
      class="bg-edit p-6"
    >
      <div
        class="flex flex-col gap-3"
      >
        <div
          class="bg-white border border-dial-gray shadow-md"
        >
          <div
            class="flex flex-row flex-wrap gap-x-8 p-4 items-center"
          >
            <div
              class="inline-flex gap-x-2 items-center text-dial-gray-dark font-bold self-start"
            >
              <img
                alt="Logo for: user1"
                class="rounded-full w-8"
                src="https://ui-avatars.com/api/name=user1&background=random"
              />
              user1
            </div>
            <div
              class="flex-1 text-justify"
            >
              <p>
                1
              </p>
            </div>
            <button
              class="delete-button gap-x-1.5"
              data-testid="delete-button"
              type="button"
            >
              <svg
                class="text-sm"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
                />
              </svg>
              <span
                class="text-sm"
              >
                Delete
              </span>
            </button>
          </div>
          <div
            class="pl-10 p-3 flex flex-col gap-3"
          >
            <div
              class="bg-white border border-dial-gray shadow-md border-l-4"
            >
              <div
                class="flex flex-row flex-wrap gap-x-8 p-4 items-center"
              >
                <div
                  class="inline-flex gap-x-2 items-center text-dial-gray-dark font-bold self-start"
                >
                  <img
                    alt="Logo for: user2"
                    class="rounded-full w-8"
                    src="https://ui-avatars.com/api/name=user2&background=random"
                  />
                  user2
                </div>
                <div
                  class="flex-1 text-justify"
                >
                  <p>
                    2
                  </p>
                </div>
                <button
                  class="delete-button gap-x-1.5"
                  data-testid="delete-button"
                  type="button"
                >
                  <svg
                    class="text-sm"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
                    />
                  </svg>
                  <span
                    class="text-sm"
                  >
                    Delete
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="cancel-button"
          data-testid="close-button"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the CommentsList component. Should match snapshot without comments. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-28"
  >
    <div
      class="card-title text-dial-gray-dark mb-3"
    >
      Comments
    </div>
    <div
      class="bg-edit p-6"
    >
      There are no comments
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="cancel-button"
          data-testid="close-button"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the CommentsList component. Should open ConfirmActionDialog after user clicks "Delete" button. 1`] = `
<div
  aria-labelledby="headlessui-dialog-title-:r2:"
  aria-modal="true"
  class="fixed z-100"
  data-headlessui-state="open"
  data-testid="confirm-action-dialog"
  id="headlessui-dialog-:r0:"
  role="dialog"
>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center p-4"
  >
    <div
      class="
 inline-block w-4/5 md:w-3/5 lg:w-3/5 xl:w-1/2 p-8
 transform bg-white shadow-xl rounded-2xl p-8 ease-out duration-300 opacity-0 scale-95"
      data-headlessui-state="open"
      id="headlessui-dialog-panel-:r1:"
    >
      <h2
        class="card-title pb-3 text-xl xl:text-2xl text-dial-sapphire"
        data-headlessui-state="open"
        id="headlessui-dialog-title-:r2:"
      >
        <span
          data-testid="title"
        >
          Deleting a comment
        </span>
      </h2>
      <span
        class="flex flex-col pb-3"
        data-testid="message"
      >
        Are you sure you want to delete this comment?
      </span>
      <div
        class="flex justify-center sm:justify-end gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="confirm-button"
          type="button"
        >
          Confirm
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
